### DISCUSSION TITLE
push to GIN stops working with "Could not read from remote repository" error

### DISCUSSION THREAD

**Discussion by alecristia** [LAAC-LSCP/ChildProject#439] [Q&A] [answered]
I've been working on a [YODA analysis repo](https://gin.g-node.org/LAAC-LSCP/tsi2017_input) that has a [nested dataset](https://gin.g-node.org//LAAC-LSCP/tsimane2017/commit/b758a549f9dda115182d61d4ed08b4346c85ee74) from multiple computers. All went well for a few days.

Just now, in a computer from which I successfully pushed in the past onto GIN, my request to push yielded:
> (ChildProjectVenv) (base) MacBook-Air-2:tsi2017_input acristia$ datalad push --to gin
Push to 'gin':  25%|████████████████████████████████                                                                                                | 1.00/4.00 [00:00<00:00, 4.41k Steps/s]/Users/acristia/.ssh/config: line 21: Bad configuration option: username
/Users/acristia/.ssh/config: terminating, 1 bad configuration options
Update availability for 'gin':  75%|████████████████████████████████████████████████████████████████████████████████████                            | 3.00/4.00 [00:00<00:00, 10.1k Steps/s]CommandError: 'git -c diff.ignoreSubmodules=none fetch gin git-annex' failed with exitcode 128 under /Users/acristia/Documents/gitrepos/tsi2017_input                                       
/Users/acristia/.ssh/config: line 21: Bad configuration option: username
/Users/acristia/.ssh/config: terminating, 1 bad configuration options
ConnectionOpenFailedError: 'ssh -fN -o ControlMaster=auto -o ControlPersist=15m -o ControlPath=/Users/acristia/Library/Caches/datalad/sockets/5b8c2a72 git@gin.g-node.org' failed with exitcode 255 [Failed to open SSH connection (could not start ControlMaster process)]
fatal: Could not read from remote repository.
Please make sure you have the correct access rights
and the repository exists.

What I checked:

- both the YODA repo & nested repo are accessible on GIN by me, so they exist & I have access rights
- I had successfully pushed in the past
- a search for this error in the discussions led me to #251 , but solutions there don't apply here

I suspect the error comes from the fact that I modified my `.ssh/config` file, though looking at it, I'm not sure why my changes affect this. I have a recollection that whenever you change your ssh config, your SSH public key changes. So I'll try updating that in GIN

**Comment by alecristia** [comment]
hmm no, when I go to [GIN > Your parameters > SSH keys](https://gin.g-node.org/user/settings/ssh) and try to add my ssh pub key as a new key, I get "Le contenu de la clé publique a été utilisée."

**Comment by alecristia** [comment]
oh something else that is probably not relevant:

The problem arose in a mac that I suspect has a git-annex problem that I haven't looked into. The files that have been modified by the commits are NOT annexed as far as I remember, but if you know how I could check, I could look into that

**Comment by LoannPeurey** [comment]
the error message strongly suggests a problem with the `.ssh/config` file like you pointed out.
Can you do any ssh operation on that computer (like `ssh oberon`) ?

And what is the content of the ocnfig file?

**Reply by alecristia** [reply]
you're on the right track! yes, my ssh is broken -- I'll DM you to see if you can see what I need to fix. Thank you!

**Reply by LoannPeurey** [reply, marked as answer]
The ssh config file had an error : using `Username` instead of `User` . replacing it resolved the issue
